prometheus/common/alert-rules.d/crunchy-alert-rules-etcd.yml.example

```diff
@@ -10,11 +10,58 @@ groups:
 
 ########## ETCD EXPORTER RULES ##########
 
+#### IMPORTANT NOTE: These alerts only support monitoring a single etcd cluster per prometheus instance ####
+
+# Ensure there is at least one etcd node reporting metrics
+  - alert: ETCDAbsent
+    expr: absent(etcd_server_has_leader)
+    for: 10s
+    labels:
+      service: etcd
+      severity: critical
+      severity_num: 300
+    annotations:
+      description: 'No etcd metric endpoints found'
+
+# Ensure there are an odd number of etcd nodes providing metrics
+  - alert: ETCDOddCount
+    expr: (count(etcd_server_has_leader) % 2) != 1
+    for: 10s
+    labels:
+      service: etcd
+      severity: critical
+      severity_num: 300
+    annotations:
+      description: 'An odd number of etcd nodes has not been found.'
+
+# Ensure there is a leader in the etcd cluster
+  - alert: ETCDLeader
+    expr: max(etcd_server_has_leader) < 1
```

Comment:
Also for understanding, this will only fire if all nodes have no leader, but will not first if one node says it has a leader and the other ones do not? (Presumably in this case, an election has just occurred, so the node reporting true is the new leader and we just checked at the wrong time before that status had propagated elsewhere?)

Reply:
Correct, it will only fire if all nodes have no leader, which means leader election failed. If one of them is disconnected and the others are still reporting a leader, this specific metric will not fire. If a cluster is missing, there are other metrics to monitor for that.